test/cypress/integration/components/filter-blog-posts.js

```diff
@@ -0,0 +1,306 @@
+    filter.clickTopic( 'Mortgages' );
+    // Then the list of topics should show only tags that contain the word "mortgage"
+    blog.resultsContent().should( 'contain', 'mortgage' );
+    // And when I select a topic in the list
+    filter.clickTopic( 'Servicemembers' );
+    // And I click "Apply filters" button
+    blog.applyFilters();
+    // Then I should see only results tagged with the selected topic
+    blog.notification().should( 'be.visible' );
+    blog.resultsContent().should( 'contain', 'service' );
+    // And the page url should contain "topics=mortgages"
+    cy.url().should( 'include', 'topics=mortgages' );
+    // And the page url should contain "topics=servicemembers"
+    cy.url().should( 'include', 'topics=servicemembers' );
+  } );
+  it( 'Select category and topic', () => {
+    // When I select a checkbox in the Category list
+    filter.checkCategoryId( 'policy_compliance' );
+    // When I select a checkbox in the Topic list
+    filter.clickTopic( 'Students' );
+    // And I click "Apply filters" button
+    blog.applyFilters();
+    // Then I should see only results that are both in the selected category and tagged with the selected topic
+    blog.notification().should( 'be.visible' );
+    blog.resultsContent().should( 'contain', 'student' );
+    blog.resultsContent().should( 'contain', 'policy' );
+    // And the page url should contain "categories=policy_compliance"
+    cy.url().should( 'include', 'categories=policy_compliance' );
+    // And the page url should contain "topics=students"
+    cy.url().should( 'include', 'topics=students' );
+  } );
+  it( 'Clear filters', () => {
+    // When I select the last checkbox in the Category list
+    filter.checkCategoryName( 'Info for consumers' );
+    // When I select a checkbox in the Topic list
+    filter.clickTopic( 'Consumer complaints' );
+    // And I click "Apply filters" button
+    filter.apply();
+    // Then the page url should contain "categories=info-for-consumers"
+    cy.url().should( 'include', 'categories=info-for-consumers' );
+    // And the page url should contain "topics=consumer-complaints"
+    cy.url().should( 'include', 'topics=consumer-complaints' );
+    // Then I should see only results that are both in the selected category and tagged with the selected topic
+    blog.resultsContent().should( 'contain', 'consumer' );
+    // And when I click "Show filters"
+    filter.show();
+    // And when I click "Clear filters"
+    filter.clear();
+    // Then the page url should not contain "categories=info-for-consumers"
+    cy.url().should( 'not.include', 'categories=info-for-consumers' );
+    // And the page url should not contain "topics=consumer-complaints"
+    cy.url().should( 'not.include', 'topics=consumer-complaints' );
+    // And when I click "Apply filters"
+    filter.apply();
+    // Then I should see the full list of results
+    blog.notification().should( 'be.visible' );
```

Comment:
Clarifying my previous suggestion in this area:

There's no reason to click "Apply filters" here, if this test is testing the "Clear filters" button, and also, we should be testing that after clearing the filters there is no visible notification.

```suggestion
    // Then I should see the full list of results
    blog.notification().should( 'not.be.visible' );
```

Reply:
Done. I made this change to remove applying filters after clearing filters and update the notification.